### PR TITLE
Add access to upstreamfilterstate in WasmStateWrapper

### PR DESCRIPTION
**Pull request by gargnupur**
For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)
Add access to upstreamfilterstate in WasmStateWrapper 

Description: 
Risk Level:
Ref:  https://github.com/istio/istio/issues/20802
Testing: Tested E2E in istio/proxy
Docs Changes:
Release Notes:
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: source/extensions/common/wasm/context.cc [hunk 1]
```diff
@@ -348,32 +348,29 @@ WasmResult serializeValue(Filters::Common::Expr::CelValue value, std::string* re
...
-          const WasmState& result = connection_filter_state_->getDataReadOnly<WasmState>(value);
-          return google::api::expr::runtime::CelValue::CreateBytes(&result.value());
-        }
-      } catch (const EnvoyException& e) {
-        return {};
-      }
-      return {};
     }
+
+    if (upstream_connection_filter_state_ &&
```
Comment (kyessenov) [inline source/extensions/common/wasm/context.cc]: What was the outcome of the discussion for the fallback of the filter state to upstream connection filter state?

Comment (gargnupur) [reply source/extensions/common/wasm/context.cc]: Don't think we discussed that...
https://github.com/envoyproxy/envoy/pull/9202/files PR added sustaining filterstate to downstream connection lifespan and then I added this https://github.com/envoyproxy/envoy/pull/9896 to share filterstate between upstream and downstream filters.
I see your point that we will suffer from the problem, what if upstream connection closes before downstream connection, then in that case upstream filterstate will be invalid. Is it possible for that to happen?

Comment (kyessenov) [reply source/extensions/common/wasm/context.cc]: It's a shared_ptr so the state will outlive the upstream connection. But we should be careful to clear it in the wasm context eventually.

Comment (kyessenov) [reply source/extensions/common/wasm/context.cc]: My question was about filter_state itself falling through to upstream connection filter state. E.g. make it try HTTP, downstream TCP, then upstream TCP.

Comment (gargnupur) [reply source/extensions/common/wasm/context.cc]: No, didnt try to make filter_state itself fall through to upstream TCP

Comment (gargnupur) [reply source/extensions/common/wasm/context.cc]: Talked offline.. will simplify in a different pr..

File: source/common/tcp_proxy/tcp_proxy.cc [hunk 2]
```diff
@@ -239,6 +239,12 @@ void Filter::initialize(Network::ReadFilterCallbacks& callbacks, bool set_connec
   getStreamInfo().setDownstreamLocalAddress(read_callbacks_->connection().localAddress());
   getStreamInfo().setDownstreamRemoteAddress(read_callbacks_->connection().remoteAddress());
   getStreamInfo().setDownstreamSslConnection(read_callbacks_->connection().ssl());
+  read_callbacks_->connection().streamInfo().setDownstreamLocalAddress(
```
Comment (jplevyak) [inline source/common/tcp_proxy/tcp_proxy.cc]: Is this being upstreamed?

Comment (gargnupur) [reply source/common/tcp_proxy/tcp_proxy.cc]: I will after getting feedback from our team...

Comment (gargnupur) [reply source/common/tcp_proxy/tcp_proxy.cc]: @jplevyak : John removed tcp proxy change her and added to envoy only first, will merge here after that... https://github.com/envoyproxy/envoy/pull/9949

Can you please help merge this PR?

**Pull request merged**